std/bitmanip.d

```diff
@@ -56,14 +56,26 @@ import std.range.primitives;
 public import std.system : Endian;
 import std.traits;
 
-private string myToString(ulong n)
+private string myToString(ulong n) pure @safe
 {
-    import core.internal.string : UnsignedStringBuf, unsignedToTempString;
-    UnsignedStringBuf buf;
-    auto s = unsignedToTempString(n, buf);
-    return cast(string) s ~ (n > uint.max ? "UL" : "U");
+    static char[] generateStr(ulong n) pure
```

Comment:
Do you really need the second function? This seems to work for me:

```d
string myToString(ulong n) pure @safe
{
    import core.internal.string : UnsignedStringBuf, unsignedToTempString;
    UnsignedStringBuf buf;
    auto s = unsignedToTempString(n, buf);
    return s ~ (n > uint.max ? "UL" : "U");
}
```

Reply:
first thing I tried was returning char[] and then I thought I don’t need to for this case and I didn’t try your idea. But not really any reason except I thought that wouldn’t work. I’ll switch to your idea

Reply:
OK, I updated to just remove the cast, and add pure/safe.